ocean_provider/validation/RBAC.py

```diff
@@ -0,0 +1,170 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import json
+from ocean_lib.common.agreements.service_types import ServiceTypesIndices
+from ocean_lib.web3_internal.transactions import sign_hash
+from ocean_provider.exceptions import RequestNotFound
+from ocean_provider.utils.basics import get_provider_wallet
+from ocean_provider.utils.util import msg_hash
+from typing import Optional
+from abc import ABC
+
+
+class RBACValidator(ABC):
+    def __init__(
+        self,
+        request_name=None,
+        request=None,
+        payload: dict = None,
+        assets: Optional[list] = None,
+        algorithms: Optional[list] = None,
+    ):
+        self._request = request
+        self._payload = payload if payload else dict()
+        if not request:
+            raise RequestNotFound("Request not found.")
+        action_mapping = {
+            "EncryptRequest": "encryptUrl",
+            "InitializeRequest": "initialize",
+            "DownloadRequest": "access",
+            "ComputeRequest": "compute",
+            "ComputeStartRequest": "compute",
+        }
+        self._action = action_mapping[request_name]
+        self._credentials = {
+            "type": "address",
+            "address": get_provider_wallet().address,
+        }
+        self._component = "provider"
+        self._provider_address = get_provider_wallet().address
+        self._assets = assets if assets else []
+        self._algorithms = algorithms if algorithms else []
+
+    @property
+    def credentials(self):
+        return self._credentials
+
+    @property
+    def provider_address(self):
+        return self._provider_address
+
+    @property
+    def component(self):
+        return self._component
+
+    @property
+    def action(self):
+        return self._action
+
+    @property
+    def assets(self):
+        return self._assets
+
+    @property
+    def request(self):
+        return self._request
+
+    @property
+    def algorithms(self):
+        return self._algorithms
+
+    @property
+    def additional_inputs(self):
+        return self._payload["additionalInputs"]
+
+    def fails(self):
+        return False
+
+    def get_compute_dict(self):
+        return {
+            "dids": [
```

Comment:
you can start with the rest of the dictionary, then simply add compute_dict["dids"] = self.get_dids() to avoid repetition

Reply:
They are different dicts. The `serviceId` is different in this case, that's why I created 2 functions and `get_compute_dict` is just for the compute service.